src/renderer/rendered_layer.rs

```diff
@@ -211,3 +211,9 @@ fn build_silhouette(regions: &[PixelRect<f32>]) -> (Path, Rect) {
 
     (silhouette, bounding_rect)
 }
+
+fn rect_to_round_rect_path(rect: &Rect, settings: &RendererSettings) -> Path {
+    let mut path = Path::new();
+    path.add_round_rect(*rect, (settings.floating_corner_radius, settings.floating_corner_radius), None);
```

Comment:
I think we at least need to take the scale factor into account here, otherwise the corners will be rounded differently when dragging the window between screens of different scale factors.

But perhaps it should even be a percentage of the line height, so that it can deal with changing the font size properly?

Reply:
Yes, I can change the code to take into account the scale factor, but a percentage of the line height should be a better solution. I haven't written a single line of rust code before, though, so I'm not quite sure how to get the current line height correctly.

Reply:
The `grid_renderer::grid_scale` contains the current line height, already multiplied by the scale factor https://github.com/neovide/neovide/blob/000b75c0f3bcfb602967cc5c2e8a9d584be06a02/src/renderer/grid_renderer.rs#L22

And `PixelSize<f32>::new(1.0, 1.0) * floating_corner_radius * grid_scale` should give you a pixel size as a fraction of the grid size when `floating_corner_radius` is less than one.  This gives different values for the X and Y axis though, so it might be better to just use the height.

Reply:
But there’s another problem here: when a floating window has two rectangles that are nested, two shadows appear(eg. `folke/noice.nvim`'s cmdline popup):
<img width="716" alt="image" src="https://github.com/user-attachments/assets/e5f055bb-47d3-4c8e-9df1-fe05fc3e4270">

However, this should be a new issue.